### PR TITLE
Fixed ImageDataGenerator.flow.

### PR DESCRIPTION
**Pull request by jnphilipp**
Every batch of saved images got overridden by the next batch. The while-loop while-loop didn't finish after the last batch and continued endless.

### CODE REVIEW

**Comment by fchollet** [comment]
> Every batch of saved images got overridden by the next batch. 

OK to fix this

> The while-loop while-loop didn't finish after the last batch and continued endless.

That's by design.

**Comment by jnphilipp** [comment]
> That's by design.

Okay, but than does the example code make no sense:
```
for X_batch, Y_batch in datagen.flow(X_train, Y_train):
        loss = model.train(X_batch, Y_batch)
```
Cause this will run endless to.

**Comment by awhitesong** [comment]
I think the while-loop in ImageDataGenerator has been changed to be used with fit_generator() which needs an endless loop. The commit says "realtime-augmentation improvements" although I'm not sure of the reasons how it improved realtime augmentation. I used model.train_on_batch() earlier, but with fresh pull, the code didn't work(due to endless while loop ofcourse). So modifying my code instead of ImageDataGenerator seems to help for now:
    
    bs = 128
    for X_train, y_train in get_train_chunk():
        b = 0
		nb_batches = int(math.ceil(float(X_train.shape[0])/bs))
		y_train = np_utils.to_categorical(y_train,nb_classes)
		for X_batch, y_batch in datagen.flow(X_train, y_train, batch_size=bs):
			if b >= nb_batches:
				break
			loss, accuracy = model.train_on_batch(X_batch, y_batch,accuracy=True)
            b += 1

**Comment by fchollet** [comment]
I fixed the documentation.
